Backpack-SwiftUI/Switch/Classes/BPKSwitch.swift

```diff
@@ -0,0 +1,112 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSwitch: View {
+    public enum Size {
+        case small, large
+    }
+    
+    @Binding private var isOn: Bool
+    private var size: Size
+    
+    public init(isOn: Binding<Bool>, size: Size = .large) {
+        self._isOn = isOn
+        self.size = size
+    }
+    
+    public var body: some View {
+        Toggle(isOn: $isOn) {
+            Group {}
+        }.toggleStyle(ColoredToggleStyle(size: size))
+        
+    }
+}
+
+private extension BPKSwitch.Size {
+    var cornerRadius: CGFloat {
+        self == .large ? 16 : 10
+    }
+    
+    var containerSize: CGSize {
+        self == .large ? CGSize(width: 52, height: 32) : CGSize(width: 40, height: 20)
+    }
+    
+    var thumbSize: CGFloat {
+        self == .large ? 28 : 16
+    }
+}
```

Comment:
None of this could be 'standardized' into Backpack spacing tokens I would assume?

Reply:
This was removed